src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health = target.health - 50;
+
+      if (target.health <= 0) {
+        Animal.removeDeadAnimals();
```

Comment:
Consider whether calling `Animal.removeDeadAnimals()` immediately after a `Herbivore`'s health reaches zero is necessary. If the task does not specify immediate removal, you might want to handle this differently, such as periodically removing dead animals or doing so at a different point in the program.